config/paths.js

```diff
@@ -3,12 +3,34 @@
 const path = require('path');
 const fs = require('fs');
 const url = require('url');
+const cosmiconfig = require('cosmiconfig');
 
 // Make sure any symlinks in the project folder are resolved:
 // https://github.com/facebookincubator/create-react-app/issues/637
 const appDirectory = fs.realpathSync(process.cwd());
 const resolveApp = relativePath => path.resolve(appDirectory, relativePath);
 
+// We look for configration in files supported by cosmiconfig by default:
+// https://github.com/davidtheclark/cosmiconfig
+const explorer = cosmiconfig('elmapp');
+const result = explorer.searchSync(appDirectory);
+const config = result ? result.config : loadElmJson();
+
+// WARNING:
+// We support config in elm.json only for legacy reasons.
+// elm-package removes the settings, so this will be removed in the future.
+function loadElmJson() {
+  try {
+    const elmJson = require(resolveApp('elm.json'));
+    if (elmJson.homepage || elmJson.proxy) {
+      return elmJson;
```

Comment:
How about printing a warning here as well, suggesting to use a config file instead of this? People _will_ use `elm-package` to install packages, and a printed warning will help people figuring out why these settings disappear, and how they can fix it.

Reply:
Good idea! I will add a deprecation warning for `elm-app` commands.